src/patient-registration/patient-registration.component.tsx

```diff
@@ -58,6 +60,7 @@ export interface FormValues {
   isDead: boolean;
   deathDate: string;
   deathCause: string;
+  relationships: any[];
```

Comment:
Is this `Array<{ relationshipType: string; personA: string; personB: string; }>`?

Reply:
Good catch. But the fix is not that straightforward. There's a problem with the design which I'm still trying to figure out a better design.

The problem is this, the relationships structure actually looks like this

```typescript
type Relationship = {
    relationshipType: string;
    personA: string;
    personB: string;
}
```

But these values are gotten from different form fields and some computation needs to be done before setting what is `personA` and what is `personB`. 

So in the form, when a person is selected, I save it in a `relationship.relatedPerson` object. See https://github.com/openmrs/openmrs-esm-patient-registration/pull/58/files#diff-1d41b748d4ef5239d089dfa8b37300103e48c4fad1901d624ef528d55cd86312R85

And when a relationship type is selected, I store this information inside a `relationship.relationship`. And this value is actually the `uuid` of the relationship joined to the `direction` as one relationship type can either be from A to B or from B to A. See https://github.com/openmrs/openmrs-esm-patient-registration/pull/58/files#diff-1d41b748d4ef5239d089dfa8b37300103e48c4fad1901d624ef528d55cd86312R103

Now here's the problem. Both `relatedPerson` and `relationship` are not part of the schema of the `relationship` field from Formik. If you see here https://github.com/openmrs/openmrs-esm-patient-registration/pull/58/files#diff-dcda704e8d2079cede6aceadeb259483396a972793aa757f1619b2d6723e0830R422

I am extracting the relationship information from the request using 

```typescript
const { relatedPerson, relationship } = tmp;
const relationshipType = relationship.split('/')[0];
const direction = relationship.split('/')[1];
```

So if I specify a type on line 63 as you commented, it breaks this

`const { relatedPerson, relationship } = tmp;` because `relatedPerson` and `relationship` are not part of the type and actually shouldn't be part of the type declaration but I'm not yet sure how to set `personA` and `personB` on the fly before form submit so as to avoid doing `const {relatedPerson, relationship} = tmp;`. 

Do you get the problem I'm trying to describe? If so do you have a suggestion for how we can handle this?